addons/l10n_id/models/account_move.py

```diff
@@ -0,0 +1,158 @@
+        ])
+        return invoices._l10n_id_update_payment_status()
+
+    def _l10n_id_update_payment_status(self):
+        """ Starts by fetching the QR statuses for the invoices in self, then update said invoices based on the statuses """
+        qr_statuses = self._l10n_id_get_qris_qr_statuses()
+        return self._l10n_id_process_invoices(qr_statuses)
+
+    def _l10n_id_get_qris_qr_statuses(self):
+        """
+        Query the API in order to get updated information on the status of each QR codes linked to the invoices in self.
+        If the QR has been paid, only the paid information is returned.
+
+        :return: a list with the format:
+            {
+                invoice: {
+                    'paid': True,
+                    'qr_statuses': [],
+                },
+                invoice: {
+                    'paid': False,
+                    'qr_statuses': [],
+                }
+            }
+        """
+        result = {}
+        for invoice in self:
+            paid = False
+            unpaid_data = []
+            paid_data = []
+            # Looping to make requests is far from ideal, but we have no choices as they don't allow getting multiple QR result at once.
+            # Ensure to loop in reverse and check from the most recent QR code.
+            for qr_invoice in reversed(invoice.l10n_id_qris_invoice_details):
+                status_response = invoice.partner_bank_id._l10n_id_qris_fetch_status(qr_invoice)
+                if status_response['data'].get('qris_status') == 'paid':
+                    paid_data.append(status_response['data'])
+                    paid = True
+                    break  # For paid invoices, we will only need the detail of the paid QR. The remaining will be discarded a bit later.
+                else:
+                    unpaid_data.append(status_response['data'])
+            result[invoice.id] = {
+                'paid': paid,
+                'qr_statuses': paid_data if paid else unpaid_data,
+            }
+        return result
+
+    def _l10n_id_process_invoices(self, invoices_statuses):
+        """
+        Receives the list of invoices and their statuses, and update them using it.
+        For peid invoices we will register the payment and log a note, while for unpaid ones we will discard expired
+        QR data and keep the non-expired ones for the next run.
+        """
+        jakarta_now = fields.Datetime.context_timestamp(self.with_context(tz='Asia/Jakarta'), fields.Datetime.now())
+        paid_invoices = self.env['account.move']
+        paid_messages = {}
+        for invoice in self:
+            statuses = invoices_statuses.get(invoice.id)
+            # Paid invoice: we simply prepare a message to notify of the payment with details if possible.
+            if statuses['paid']:
+                paid_status = next(status for status in statuses['qr_statuses'] if status['qris_status'] == 'paid')
```

Comment:
In theory yes, unless someone paid the invoice twice using two different QR.
I guess we could just get the [0] here anyway since after the last changes these would all be paid